superset/assets/src/SqlLab/components/QueryAutoRefresh.jsx

```diff
@@ -38,12 +38,23 @@ class QueryAutoRefresh extends React.PureComponent {
   }
   stopwatch() {
     // only poll /superset/queries/ if there are started or running queries
+    const self = this;
     if (this.shouldCheckForQueries()) {
       const url = `/superset/queries/${this.props.queriesLastUpdate - QUERY_UPDATE_BUFFER_MS}`;
-      $.getJSON(url, (data) => {
-        if (Object.keys(data).length > 0) {
-          this.props.actions.refreshQueries(data);
-        }
+      $.ajax({
+        dataType: 'json',
+        url,
+        timeout: 3000,
```

Comment:
this line will make all long query (>3 seconds) abort.

Reply:
Okay, I will remove the timeout option.